### PR TITLE
PLANNER-1587: Move Admin Service and Solver

### PR DESCRIPTION
**Pull request by cuijulian**
**WannabeSolverManager**
- Move `WannabeSolverManager` to Spring Boot
- Implement `solve()`, `terminate()`, `getRoster()`, and `getScoreDirector()` methods
- Add solver tests

**IndictmentUtils**
- Move `IndictmentUtils` to Spring Boot

**ShiftService**
- Implement `Shift` CRUD methods in `ShiftController.java` and `ShiftService.java`
- Add `ShiftRepository.java` for database persistence
- Add unit and integration tests for `Shift` CRUD methods

**ShiftRosterView**
- Implement `ShiftRosterView` GET methods in `RosterController.java` and `RosterService.java`
- Add integration tests for `ShiftRosterView` GET methods

**AvailabilityRosterView**
- Implement `AvailabilityRosterView` GET methods in `RosterController.java` and `RosterService.java`
- Add integration tests for `AvailabilityRosterView` GET methods

**Publish and Provision**
- Implement `publishAndProvision()` in `RosterController.java` and `RosterService.java`
- Add integration test for `publishAndProvision()`

**AdminService**
- Implement `resetApplication()` in `AdminController.java` and `AdminService.java`
- Add integration test for reset

**Minor Fixes**
- Add `Pagination` class to display entities in blocks
- Fix URL paths for certain `update` methods for consistency with old backend
- Add `OptaPlannerJacksonModule` to serialize score
- Reformat `Optional` object checks

### CODE REVIEW

File: employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterGenerator.java [hunk 1]
```diff
@@ -418,13 +418,6 @@ public void checkForExistingData() {
 
     @Transactional
     public void setUpGeneratedData() {
-        // Check if Tenant entities already exist before generating data
```
Comment (Christopher-Chianelli) [inline employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterGenerator.java]: Why did you delete this? This ensures that the data is not regenerated if you reboot the application with a persistent database.

File: employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterService.java [hunk 2]
```diff
@@ -50,12 +116,294 @@ public RosterState getRosterState(Integer tenantId) {
...
+
+        List<Shift> shiftList = shiftRepository.filterWithSpots(tenantId, spotSet,
+                                                                startDate.atStartOfDay(timeZone).toOffsetDateTime(),
+                                                                endDate.atStartOfDay(timeZone).toOffsetDateTime());
+
+        Map<Long, List<ShiftView>> spotIdToShiftViewListMap = new LinkedHashMap<>(spotList.size());
+        // TODO FIXME race condition solverManager's bestSolution might differ from the one we just fetched, so the
+        //  score might be inaccurate
+        Roster roster = solverManager.getRoster(tenantId);
+        if (roster == null) {
```
Comment (Christopher-Chianelli) [inline employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterService.java]: Inconsistent use of Yoda; you used a Yoda condition above, but this condition is not Yoda.

File: employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterService.java [hunk 3]
```diff
@@ -50,12 +116,294 @@ public RosterState getRosterState(Integer tenantId) {
...
+        List<EmployeeAvailability> employeeAvailabilityList = employeeAvailabilityRepository.findAllByTenantId(tenantId)
+                .stream()
+                .map(ea -> ea.inTimeZone(zoneId))
+                .collect(Collectors.toList());
+        List<Shift> shiftList = shiftRepository.findAllByTenantId(tenantId)
+                .stream()
+                .map(s -> s.inTimeZone(zoneId))
+                .collect(Collectors.toList());
+
+        // TODO fill in the score too - do we inject a ScoreDirectorFactory?
```
Comment (Christopher-Chianelli) [inline employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterService.java]: Remove TODO; We already set the score (see below)

File: employee-rostering-backend/src/main/java/org/optaweb/employeerostering/OptaWebEmployeeRosteringApplication.java [hunk 4]
```diff
@@ -16,12 +16,20 @@
...
+import org.optaplanner.persistence.jackson.api.OptaPlannerJacksonModule;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class OptaWebEmployeeRosteringApplication {
 
+    @Bean
+    public Module OptaplannerModule() {
```
Comment (Christopher-Chianelli) [inline employee-rostering-backend/src/main/java/org/optaweb/employeerostering/OptaWebEmployeeRosteringApplication.java]: rename to `getOptaplannerModule()` if possible.

File: employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterGenerator.java (hunk 1 above)
Comment (cuijulian) [reply employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterGenerator.java]: This check was already done in the `checkForExistingData()` method above, which is called when the application starts.
